ci/patch_bazel_windows/compile.yml

```
@@ -0,0 +1,99 @@
+# Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+parameters:
+- name: final_job_name
+
+jobs:
+
+- job: patch_bazel_pre_check
+  pool:
+    name: linux-pool
+  steps:
+  - checkout: self
+  - bash: |
+      set -euo pipefail
+
+      CACHE_KEY="$(md5sum $(find ci/patch_bazel_windows -type f) | md5sum | awk '{print $1}')"
+      TARGET="patch_bazel_windows/bazel-$CACHE_KEY.zip"
+      TARGET_URL="https://daml-binaries.da-ext.net/$TARGET"
+
+      if [ "200" = "$(curl -Is "$TARGET_URL" | awk '{print $2}')" ]; then
+          SHOULD_RUN=false
+      else
+          SHOULD_RUN=true
+      fi
+      echo "##vso[task.setvariable variable=cache_key;isOutput=true]$CACHE_KEY"
+      echo "##vso[task.setvariable variable=should_run;isOutput=true]$SHOULD_RUN"
+      echo "##vso[task.setvariable variable=target;isOutput=true]$TARGET"
+      echo "##vso[task.setvariable variable=target_url;isOutput=true]$TARGET_URL"
+    name: out
+
+- job: patch_bazel_compile
+  dependsOn:
+    - patch_bazel_pre_check
+  variables:
+    cache_key: $[ dependencies.patch_bazel_pre_check.outputs['out.cache_key'] ]
+    bazel_base_version: 2.1.0
+  condition: and(succeeded(), eq(dependencies.patch_bazel_pre_check.outputs['out.should_run'], 'true'))
+  pool:
+    vmImage: windows-2019
+  steps:
+  - checkout: self
+  - bash: |
+      git clone https://github.com/bazelbuild/bazel.git
+      cd bazel
+      git checkout $(bazel_base_version)
+      git apply --ignore-space-change --ignore-whitespace --whitespace=nowarn < ../ci/patch_bazel_windows/patch-bazel
+  - powershell: |
+      choco install msys2 --noprogress --yes
+      choco install zip --noprogress --yes
+  - powershell: |
+      C:\tools\msys64\usr\bin\pacman -S zip --noconfirm
+  - bash: |
+      set -euo pipefail
+      cd bazel
+      bazel build src/main/cpp:client src:package-zip_jdk_minimal -c opt --stamp --embed_label $(bazel_base_version)-patched-$(cache_key)
+      /c/tools/msys64/msys2_shell.cmd -defterm -no-start -here -c "cat bazel-bin/src/main/cpp/client.exe bazel-bin/src/package_jdk_minimal.zip > bazel.exe && zip -A bazel.exe"
```

```suggestion
      # Note (MK) For some reason, the `zip` from chocolatey seems to result in 
      # a “zip file structure invalid” error. I’ve tried adding msys to PATH so the Bazel
      # rules pick up `zip` from msys but that broke other things. So for now
      # we skip the final Bazel rule to build the self-extracting exe and instead
      # call `zip` from msys separately.
      /c/tools/msys64/msys2_shell.cmd -defterm -no-start -here -c "cat bazel-bin/src/main/cpp/client.exe bazel-bin/src/package_jdk_minimal.zip > bazel.exe && zip -A bazel.exe"
```
I’m sure I’ll forget why we do this so adding my notes here seems useful :slightly_smiling_face: